beacon-chain/rpc/prysm/v1alpha1/validator/proposer_epbs_test.go

```diff
@@ -0,0 +1,99 @@
+	"context"
+	"testing"
+
+	chainMock "github.com/prysmaticlabs/prysm/v5/beacon-chain/blockchain/testing"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/cache"
+	powtesting "github.com/prysmaticlabs/prysm/v5/beacon-chain/execution/testing"
+	doublylinkedtree "github.com/prysmaticlabs/prysm/v5/beacon-chain/forkchoice/doubly-linked-tree"
+	mockSync "github.com/prysmaticlabs/prysm/v5/beacon-chain/sync/initial-sync/testing"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v5/encoding/ssz"
+	v1 "github.com/prysmaticlabs/prysm/v5/proto/engine/v1"
+	eth "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/testing/require"
+	"github.com/prysmaticlabs/prysm/v5/testing/util"
+)
+
+func TestServer_GetLocalHeader(t *testing.T) {
+	t.Run("Node is syncing", func(t *testing.T) {
+		vs := &Server{
+			SyncChecker: &mockSync.Sync{IsSyncing: true},
+		}
+		_, err := vs.GetLocalHeader(context.Background(), &eth.HeaderRequest{
+			Slot:          0,
+			ProposerIndex: 0,
+		})
+		require.ErrorContains(t, "Syncing to latest head, not ready to respond", err)
+	})
+	t.Run("ePBS fork has not occurred", func(t *testing.T) {
+		vs := &Server{
+			SyncChecker: &mockSync.Sync{IsSyncing: false},
+			TimeFetcher: &chainMock.ChainService{},
+		}
+		_, err := vs.GetLocalHeader(context.Background(), &eth.HeaderRequest{
+			Slot:          0,
+			ProposerIndex: 0,
+		})
+		require.ErrorContains(t, "EPBS fork has not occurred yet", err)
+	})
+	t.Run("Happy case", func(t *testing.T) {
+		params.SetupTestConfigCleanup(t)
+		cfg := params.BeaconConfig().Copy()
+		cfg.EPBSForkEpoch = 1
+		params.OverrideBeaconConfig(cfg)
+
+		st, _ := util.DeterministicGenesisStateEpbs(t, 1)
+		fc := doublylinkedtree.New()
+		chainService := &chainMock.ChainService{
+			ForkChoiceStore: fc,
+			State:           st,
+		}
+		payloadIdCache := cache.NewPayloadIDCache()
+		payloadId := primitives.PayloadID{1}
+		payloadIdCache.Set(params.BeaconConfig().SlotsPerEpoch, [32]byte{}, payloadId)
+
+		payload := &v1.ExecutionPayloadElectra{
+			ParentHash: []byte{1},
+			BlockHash:  []byte{2},
+			GasLimit:   1000000,
+		}
+		executionData, err := blocks.NewWrappedExecutionData(payload)
+		require.NoError(t, err)
+		kzgs := [][]byte{make([]byte, 48), make([]byte, 48), make([]byte, 48)}
+		vs := &Server{
+			SyncChecker:            &mockSync.Sync{IsSyncing: false},
+			TimeFetcher:            chainService,
+			ForkchoiceFetcher:      chainService,
+			HeadFetcher:            chainService,
+			TrackedValidatorsCache: cache.NewTrackedValidatorsCache(),
+			PayloadIDCache:         payloadIdCache,
+			ExecutionEngineCaller: &powtesting.EngineClient{
+				GetPayloadResponse: &blocks.GetPayloadResponse{ExecutionData: executionData, BlobsBundle: &v1.BlobsBundle{
+					KzgCommitments: kzgs,
+				}},
+			},
+		}
```

Comment:
Not a big of a deal but I think this is easier to read:

```suggestion
			ExecutionEngineCaller: &powtesting.EngineClient{
				GetPayloadResponse: &blocks.GetPayloadResponse{
					ExecutionData: executionData,
					BlobsBundle: &v1.BlobsBundle{
						KzgCommitments: kzgs,
					}},
			},
```